src/v/cluster/persisted_stm.cc

```diff
@@ -535,6 +535,7 @@ ss::future<> persisted_stm<T>::start() {
               next_offset);
             co_await apply_snapshot(snapshot.header, std::move(snapshot.data));
             set_next(next_offset);
+            _last_snapshot_offset = snapshot.header.offset;
```

Comment:
nit: looks like the commit message is a bit off - this is `start()`, not `take_snapshot()`, right?